aten/src/ATen/native/TensorShape.cpp

```diff
@@ -843,12 +843,9 @@ Tensor diag_embed(const Tensor& self, int64_t offset, int64_t dim1_, int64_t dim
   return result;
 }
 
-Tensor expand_symint(const Tensor& self, c10::SymIntArrayRef packed_size, bool implicit) {
-  auto size = asIntArrayRefSlow(packed_size);
-  return self.expand(size, implicit);
-}
-
-Tensor expand(const Tensor& self, IntArrayRef size, bool /*unused*/) {
+Tensor expand(const Tensor& self, c10::SymIntArrayRef sym_size, bool /*unused*/) {
+  // TODO: properly support SymInt expand
+  auto size = asIntArrayRefSlow(sym_size);
```

Comment:
I have a basic question, partially to clarify when we should do things in C++ vs. python.

One thing we mentioned before is that we can't really use python implementations for view ops to do symbolic shape compute, because we need the output tensor to share storage with the inner tensor (which requires stuff that lives in C++).

So when we're tracing with dynamic shapes and call the meta function for expand(), we won't be able to rely on a python impl for the meta function, and will need to beef up this code to actually support plumbing SymInts?

More general question: how much of the "symbolic-aware meta function" code do you think we should have implemented in python vs. C++? The extreme end would be "symint code for views in C++, everything in python?"

Reply:
I think the rules are now (1) the op cannot be correctly implemented in python and (2) the op is called directly from C++ and has symint arguments. The second is important because you obviously need a symint overload to be able to pass symints into the function at all.

Reply:
ah right. So basically, view ops, and ops that show up in derivative formulas (and maybe more that I'm not thinking of)